code/datums/components/swabbing.dm

```diff
@@ -0,0 +1,127 @@
+/*!
+
+This component is used in vat growing to swab for microbiological samples which can then be mixed with reagents in a petridish to create a culture plate.
+
+*/
+/datum/component/swabbing
+	///The current datums on the swab
+	var/list/swabbed_items
+	///Can we swab objs?
+	var/CanSwabObj
+	///Can we swab turfs?
+	var/CanSwabTurf
+	///Can we swab mobs?
+	var/CanSwabMob
+	///Callback for update_icon()
+	var/datum/callback/UpdateIcons
+	///Callback for update_overlays()
+	var/datum/callback/UpdateOverlays
```

Comment:
Use snake case on this codebase and clean these variables on component destruction